src/flows/pipes/Notification/Threshold.tsx

```diff
@@ -0,0 +1,269 @@
+    if (key === 'result' || key === 'table') {
+      return false
+    }
+
+    const columnType = results.parsed.table.getColumnType(key)
+
+    return columnType === 'time' || columnType === 'number'
+  })
+
+  const setThresholdType = type => {
+    if (!THRESHOLD_TYPES[type]) {
+      return
+    }
+
+    if (THRESHOLD_TYPES[type].format === ThresholdFormat.Range) {
+      update({
+        threshold: {
+          type,
+          field: data.threshold.field || '_value',
+          min: 0,
+          max: 100,
+        },
+      })
+    } else {
+      update({
+        threshold: {
+          type,
+          field: data.threshold.field || '_value',
+          value: 20,
+        },
+      })
+    }
+  }
+
+  const setColumn = column => {
+    update({
+      threshold: {
+        ...data.threshold,
+        field: column,
+      },
+    })
+  }
+
+  const funcDropdown = useMemo(() => {
+    const menuItems = Object.entries(THRESHOLD_TYPES).map(([key, value]) => (
+      <Dropdown.Item
+        key={key}
+        value={key}
+        onClick={setThresholdType}
+        selected={key === data.threshold?.type}
+        title={value.name}
+      >
+        {value.name}
+      </Dropdown.Item>
+    ))
+    const menu = onCollapse => (
+      <Dropdown.Menu onCollapse={onCollapse}>{menuItems}</Dropdown.Menu>
+    )
+    const menuButton = (active, onClick) => (
+      <Dropdown.Button
+        onClick={onClick}
+        active={active}
+        size={ComponentSize.Medium}
+      >
+        {THRESHOLD_TYPES[data.threshold?.type]?.name || 'Select a function'}
+      </Dropdown.Button>
+    )
+    return <Dropdown menu={menu} button={menuButton} />
+  }, [data.threshold?.type])
+
+  const columnDropdown = useMemo(() => {
+    const menuItems = numericColumns.map(key => (
+      <Dropdown.Item
+        key={key}
+        value={key}
+        onClick={setColumn}
+        selected={key === data.threshold?.field}
+        title={key}
+      >
+        {key}
+      </Dropdown.Item>
+    ))
+    const menu = onCollapse => (
+      <Dropdown.Menu onCollapse={onCollapse}>{menuItems}</Dropdown.Menu>
+    )
+    const menuButton = (active, onClick) => (
+      <Dropdown.Button
+        onClick={onClick}
+        active={active}
+        size={ComponentSize.Medium}
+      >
+        {data.threshold?.field || 'Select a numeric column'}
+      </Dropdown.Button>
+    )
+    return <Dropdown menu={menu} button={menuButton} />
+  }, [numericColumns, data.threshold?.field])
+
+  const updateMin = evt => {
+    update({
+      threshold: {
+        ...data.threshold,
+        min: evt.target.value,
+      },
+    })
+  }
+
+  const updateMax = evt => {
+    update({
+      threshold: {
+        ...data.threshold,
+        max: evt.target.value,
+      },
+    })
+  }
+
+  const updateValue = evt => {
+    update({
+      threshold: {
+        ...data.threshold,
+        value: evt.target.value,
+      },
+    })
+  }
+
+  const feet =
```

Comment:
nit: can we rename this to something a little more relevant?

Reply:
foots?

Reply:
What’s that const doing?

Reply:
a list of components that go into the footer. foots go in footers, a singular of which is a foot, but feet is the proper pluralization of the word foot